pyuvdata/uvfits.py

```diff
@@ -449,8 +454,18 @@ def read_uvfits(self, filename, antenna_nums=None, antenna_names=None,
             # stuff in columns
             ant_names = ant_hdu.data.field('ANNAME').tolist()
             self.antenna_names = []
-            for name in ant_names:
-                self.antenna_names.append(name.replace('\x00!', ''))
+            for ant_ind, name in enumerate(ant_names):
+                # Sometimes CASA writes antnames as bytes not strings.
+                # If the ant name is shorter than 8 characters, the trailing
+                # characters may not be able to be decoded.
+                # So we just ignore any non-ascii bytes in the decode.
+                if isinstance(name, bytes):
+                    ant_name_str = None
```

Comment:
I'm not sure what this line is doing?

Reply:
It's cruft from an earlier version. I'll take it out.